app/src/main/java/com/dodo/flashcards/presentation/MainModels.kt

```diff
@@ -1,7 +1,22 @@
 package com.dodo.flashcards.presentation
 
 import com.dodo.flashcards.architecture.Destination
+import com.dodo.flashcards.architecture.ViewEvent
+import com.dodo.flashcards.architecture.ViewState
+import com.google.firebase.auth.FirebaseAuth
+
+sealed interface MainViewState : ViewState {
+    object Unauthenticated : MainViewState
+    object Authenticated : MainViewState
+
+}
+
+sealed interface MainViewEvent : ViewEvent {
+    data class AuthenticateUser(val auth : FirebaseAuth) : MainViewEvent
```

Comment:
This event can probably be removed - the reason there was an event in the StravaActivityArt was because that event parsed a URI created when the application was opened from an external source (when you open the app from the Strava connection screen). 

Here, we can just inject FirebaseAuth directly into MainViewModel and in the `init` block determine authentication.